File: Python/fibonaccinumber.py
class Solution:
    def fib(self, n: int) -> int:
        answer = {}
        answer[0] = 0
        answer[1] = 1
        if n > 1: 
            for i in range(2, n+1):
                answer[i] = (answer[i-1] + answer[i-2])
        return answer[n]

File: Golang/fibonaccinumber.go
func fib(N int) int {
	fib := make([]int, 31)
	if N < 2 { return N }
	fib[0] = 0
	fib[1] = 1
	for i := 2; i <= N; i++ { fib[i] = fib[i-1] + fib[i-2] }
	return fib[N]
}
